cache_utils/decorators.py

```diff
@@ -46,16 +71,26 @@ def test(*args, **kwargs):
         cache_backend = caches['default']
 
     def _cached(func):
-        func_type = _func_type(func)
+        fn_key_str = str(
```

Comment:
this could become `str(None) = "None"` (if you explicitly pass `fn_key=None` for any reason) and then it wouldn't raise an error below.

Do you think we should maybe also add an if above for fn_key to not be falsy? Or if a user provides an argument, then maybe they check docs? But there aren't any, so the code in this case.

Reply:
Great catch! We can make a hard requirement for `fn_key` to either be a callable or a `str`, and the result of a callable `fn_key` to be a `str` to remove the `str` casting. If these requirements fail, we raise an exception. This should avoid any unexpected behavior.